libfreerdp/core/gateway/arm.c

```diff
@@ -594,9 +594,93 @@ static BOOL arm_pick_base64Utf16Field(const WINPR_JSON* json, const char* name,
  *
  */
 
+static size_t arm_parse_ipvx_count(WINPR_JSON* ipvX)
+{
+	WINPR_ASSERT(ipvX);
+	WINPR_JSON* ipAddress = WINPR_JSON_GetObjectItem(ipvX, "ipAddress");
+	if (!ipAddress || !WINPR_JSON_IsArray(ipAddress))
+		return 0;
+	return WINPR_JSON_GetArraySize(ipAddress);
+}
+
+static BOOL arm_parse_ipv6(rdpSettings* settings, WINPR_JSON* ipv6, size_t* pAddressIdx)
+{
+	WINPR_ASSERT(settings);
+	WINPR_ASSERT(ipv6);
+	WINPR_ASSERT(pAddressIdx);
+
+	if (!freerdp_settings_get_bool(settings, FreeRDP_IPv6Enabled))
+		return TRUE;
+
+	WINPR_JSON* ipAddress = WINPR_JSON_GetObjectItem(ipv6, "ipAddress");
+	if (!ipAddress || !WINPR_JSON_IsArray(ipAddress))
+		return TRUE;
+	const size_t naddresses = WINPR_JSON_GetArraySize(ipAddress);
+	for (size_t j = 0; j < naddresses; j++)
+	{
+		WINPR_JSON* adressN = WINPR_JSON_GetArrayItem(ipAddress, j);
+		if (!adressN || !WINPR_JSON_IsString(adressN))
+			continue;
+		if (!freerdp_settings_set_pointer_array(settings, FreeRDP_TargetNetAddresses,
+		                                        (*pAddressIdx)++, adressN))
+			return FALSE;
+	}
+	return TRUE;
+}
+
+static BOOL arm_parse_ipv4(rdpSettings* settings, WINPR_JSON* ipv4, size_t* pAddressIdx)
+{
+	WINPR_ASSERT(settings);
+	WINPR_ASSERT(ipv4);
+	WINPR_ASSERT(pAddressIdx);
+
+	WINPR_JSON* ipAddress = WINPR_JSON_GetObjectItem(ipv4, "ipAddress");
+	if (!ipAddress || !WINPR_JSON_IsArray(ipAddress))
+		return TRUE;
+
+	const size_t naddresses = WINPR_JSON_GetArraySize(ipAddress);
+	for (size_t j = 0; j < naddresses; j++)
+	{
+		WINPR_JSON* adressN = WINPR_JSON_GetArrayItem(ipAddress, j);
+		if (!adressN)
+			continue;
+
+		WINPR_JSON* publicIpNode = WINPR_JSON_GetObjectItem(adressN, "publicIpAddress");
+		if (publicIpNode && WINPR_JSON_IsString(publicIpNode))
+		{
+			const char* publicIp = WINPR_JSON_GetStringValue(publicIpNode);
+			if (publicIp)
```

Comment:
This results in publicIp=="" for 

```
"azureInstanceNetworkMetadata": "{\"interface\":[{\"ipv4\":{\"ipAddress\":[{\"privateIpAddress\":\"10.234.60.7\",\"publicIpAddress\":\"\"}],\"subnet\":[{\"address\":\"10.234.60.0\",\"prefix\":\"24\"}]},\"ipv6\":{\"ipAddress\":[]},\"macAddress\":\"6045BDD6FD58\"}]}"
```

which results in `freerdp_settings_set_pointer_array` failing for the privateIp on line 666 because `offset >= maxOffset` is true.

Reply:
ok, pushed a fix for empty strings